### ISSUE TITLE
Error: Cannot find module '../conf/eslint.json'

### ISSUE THREAD

**Issue by StaticNoiseLog** [Microsoft/vscode-eslint#382]
Versions:
Plugin: ESLint 1.4.3
Node module: eslint@4.15.0
Visual Studio Code: Version 1.19.1

I am on a Mac and basically did this:

- sudo npm install -g eslint
- Open Visual Studio Code with a Multi-root Workspace
- Install the vscode-eslint plugin

I now get the following error (actually it is just an Info message) on startup of VSC in its OUTPUT window. Things seem to be working, but it is a bit messy. I tried to create empty ../conf/eslint.json files in various locations, but the error remained.

`[Info  - 13:45:01] ESLint server is running.
(node:3892) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Cannot find module '../conf/eslint.json'
(node:3892) DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:3892) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): Error: Cannot find module '../conf/eslint.json'`

**Comment by dbaeumer** [comment]
@StaticNoiseLog I tried to reproduce this in my test multi root workspace but could not. Can you provide me with a GitHub repository that I can clone that demos this behavior ?

**Comment by StaticNoiseLog** [comment]
Thanks for having a look, surely not an urgent issue. By coincidence I saw that the problem seems to be related with the presence of a node_modules folder (explained at the end)

Here is how to reproduce the problem:

In an empty directory:
    git clone https://github.com/StaticNoiseLog/vscode-eslint_382.git

In Visual Studio Code:
File, Open Workspace...
../vscode-eslint_382/DEV.code-workspace
Open Aurelia/contact-manager/src/app.js

View, Output -> choose ESLint for the output

The UnhandledPromiseRejectionWarning appears once when closing app.js and twice when opening it. (Or another JS file).

**Noticed by coincidence: If this folder is removed ...**
    rm -r Aurelia/contact-manager/node_modules/ 
**... and the workspace closed and reopened, the warning is gone.**

**Comment by dbaeumer** [comment]
@StaticNoiseLog thanks. I cloned the repository, but it contains a prepopulated node_modules folder which doesn't work on my machine. Additionally the repository doesn't contain a src/app.js file. Can you please provide me with an example that contains a package.json file which I can use to install the dependencies. 

Looking at the code a little I suspect that this might be a problem of not having eslint and all its plugins either installed all globally or all locally (this is a limitation of ESLint itself and not the ESLint extension)

**Comment by Kappert1Swisscom** [comment]
Yes, I have put node_modules in Git because without it the problem does not occur. It is well possible that application is not in a working state. You do not have to run the application. Just open a JavaScript file.
The app.js should be there, but is in a subdirectory:
https://github.com/StaticNoiseLog/vscode-eslint_382/blob/master/Aurelia/contact-manager/src/app.js

Thanks for the suggestion that it may be bad installation of ESLint. I will verify this and get back to you (but might take a while).

**Comment by StaticNoiseLog** [comment]
Yes, I have put node_modules in Git because without it the problem does not occur. It is well possible that the application is not in a working state. You do not have to run the application. Just open a JavaScript file for editing.
The app.js should be there, but is in a subdirectory:
https://github.com/StaticNoiseLog/vscode-eslint_382/blob/master/Aurelia/contact-manager/src/app.js

Thanks for the suggestion that it may be bad a installation of ESLint. I will verify this and get back to you (but might take a while).

**Comment by StaticNoiseLog** [comment]
@dbaeumer Your hint was good! I had installed ESLint globally (eslint@4.15.0) and there was a local node package in this project with a dependency on an older ESLint version (eslint@2.13.1).

I used this command and (after some resistance) could uninstall gulp-eslint:
    npm uninstall gulp-eslint

This got rid of the message.
I don't know what removing these packages will break and where they came from in this Aurelia example. But I just wanted to know what the message meant and I can still start the app.

From my side, this is solved. Feel free to close the issue if it is unrelated to this plugin.
